ghost/offers/lib/application/OfferRepository.js

```diff
@@ -179,6 +187,39 @@ class OfferRepository {
         return Promise.all(offers);
     }
 
+    /**
+      * @param {import('stripe').Stripe.CouponCreateParams} coupon
+      * @param {OfferAdditionalParams} params
+      * @param {BaseOptions} [options]
+     */
+    async createFromCoupon(coupon, params, options) {
+        const {productId, currency, interval, active} = params;
+        const code = coupon.name && coupon.name.split(' ').map(word => word.toLowerCase()).join('-');
+
+        const data = {
+            active,
+            name: coupon.name,
+            code,
+            product_id: productId,
```

Comment:
I might be wrong, but I don't see the product_id and interval options in the [Stripe docs](https://stripe.com/docs/api/coupons/create)? Or maybe I'm misunderstanding something

Reply:
Yes indeed, they're not present in the Stripe Coupon (there are more data fields in the Offer object, than in the Stripe Coupon):
- Product ID is the internal Ghost Product ID
- Currency and Interval are stored outside of the Coupon in Stripe

cf. these [lines](https://github.com/TryGhost/Ghost/blob/2574409ce3fd70674f59df69c1272abfceb4b0f5/ghost/members-api/lib/repositories/MemberRepository.js#L954C1-L956C93)